kml.cc

```diff
@@ -392,6 +415,35 @@ void trk_coord(xg_string args, const QXmlStreamAttributes*)
     track_add_wpt(trk_head, trkpt);
     iargs = iargs.mid(consumed);
   }
+
+  /* The track coordinates do not have a time associated with them. This is specified by using:
+   *
+   * <TimeSpan>
+   *   <begin>2017-08-21T17:00:05Z</begin>
+   *   <end>2017-08-21T17:22:32Z</end>
+   * </TimeSpan>
+   *
+   * If this is specified, then SetCreationDate
+   */
+  if ( wpt_timespan_begin.isValid() && wpt_timespan_end.isValid() ) {
+
+	  // Count the number of Waypoints
+	  n = 0;
+	  QUEUE_FOR_EACH(&trk_head->waypoint_list, curr_elem, tmp_elem) {
+		  n++;
```

Comment:
Consider n = trk_head.rte_waypt_ct;
This is admittedly poorly named and not obvious and not the very "C++" way of doing things, but see if that plays out well.